lib/cucumber/formatter/http_io.rb

```diff
@@ -35,31 +35,34 @@ def initialize(url, https_verify_mode=nil)
         headers.each do |header, value|
           @req[header] = value
         end
-        
+
         @http = Net::HTTP.new(uri.hostname, uri.port)
-        if uri.scheme == 'https'
-          @http.use_ssl = true
-          @http.verify_mode = https_verify_mode if https_verify_mode
-        end
+        return unless uri.scheme == 'https'
+
+        @http.use_ssl = true
+        @http.verify_mode = https_verify_mode if https_verify_mode
       end
-      
+
       def write(chunk)
         @body += chunk
       end
-      
+
       def closed?
         @closed
       end
-      
+
       def flush
         @req.body = @body
         res = @http.request(@req)
-        raise "Not OK" unless Net::HTTPOK === res
+        # Note: Rubocop does not like res.kind_of?(Net::HTTPOK) neither
+        # rubocop:disable Style/CaseEquality
+        raise 'Not OK' unless Net::HTTPOK === res
```

Comment:
you can do `is_a?` combined with `casecmp?` for optimisation. But It might look a bit nastier.

I'm not sure what Net::HTTPOK returns. I'm guessing a string. If so, you might be able to do it without casecmp

Reply:
I tried `is_a?` but it makes the Rspec tests fails :/

Reply:
No longer relevant